index.js

```diff
@@ -1,24 +1,21 @@
-'use strict';
-
+const assert = require('assert');
 const AWS = require('aws-sdk');
 const elasticsearch = require('elasticsearch');
 const httpAWSES = require('http-aws-es');
 
-function createClient(options) {
-	options = options || {}; // eslint-disable-line no-param-reassign
+function createClient(options = {}) {
+	assert(typeof options === 'object', 'options must be an object');
 
 	const region = options.region || process.env.AWS_REGION || 'us-east-1';
-	const config = Object.assign(
-		{},
-		options,
-		{
-			connectionClass: httpAWSES,
-			awsConfig: new AWS.Config({
-				region,
-				credentials: options.credentials
-			})
-		}
-	);
+
+	const config = {
```

Comment:
CircleCI config for this project uses node 6, which needs to be updated to v8 to support destructuring.
https://github.com/AntonBazhal/aws-elasticsearch-client/blob/master/.circleci/config.yml#L10